homeassistant/components/haiku/fan.py

```diff
@@ -0,0 +1,170 @@
+"""Platform for Haiku integration."""
+import asyncio
+import logging
+
+from haiku import discover, fan
+
+from homeassistant.components.fan import (
+    ATTR_SPEED,
+    ATTR_SPEED_LIST,
+    ENTITY_ID_FORMAT,
+    SPEED_OFF,
+    SUPPORT_SET_SPEED,
+)
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+
+_LOGGER = logging.getLogger(__name__)
+SPEED_1 = "Speed 1"
+SPEED_2 = "Speed 2"
+SPEED_3 = "Speed 3"
+SPEED_4 = "Speed 4"
+SPEED_5 = "Speed 5"
+SPEED_6 = "Speed 6"
+SPEED_7 = "Speed 7"
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Haiku platform."""
+    devices = asyncio.run(discover.discover())
+    addlist = []
+    if devices[1] == [] or devices is False:
+        return False
+    for devicesrun in range(len(devices[1])):
+        afan = ""
+        fanlist = (devices[1])[devicesrun]
+        afan = HaikuFan(hass, fanlist[2], fanlist[3], fanlist[1])
+        afan.name = fanlist[0]
+        addlist.append(afan)
+    add_entities(addlist)
+
+
+class HaikuFan(Entity):
+    """Representation of a fan."""
+
+    _name = None  # type: str
+
+    def __init__(self, hass, uid, cip, hstr):
+        """Initialize the fan."""
+        self._hass = hass
+        self._cip = cip
+        self._hstr = hstr
+        self._state = None
+        self._entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, self.name, hass=hass
+        )
+        self._unique_id = uid
+        self._speed_list = []
+        self._supported_features = 0
+        self._state = STATE_ON
+        self._supported_features = SUPPORT_SET_SPEED
+        self._speed_list = [
+            SPEED_OFF,
+            SPEED_1,
+            SPEED_2,
+            SPEED_3,
+            SPEED_4,
+            SPEED_5,
+            SPEED_6,
+            SPEED_7,
+        ]
```

Comment:
Only the speeds available in the fan entity integration are allowed at the current time. We're working on changing interface to percentage, but that's not ready yet.
https://github.com/home-assistant/core/blob/7d4a9ac65adb5a31bf254d3117c3ed174ae4f10a/homeassistant/components/fan/__init__.py#L40-L43

Reply:
Have you changed the entity model yet?

Reply:
No.